### ISSUE TITLE
Bazel crashes with java.lang.NullPointerException when vendoring dependencies

### ISSUE THREAD

**Issue by mortenmj** [bazelbuild/bazel#23300]
### Description of the bug:

Running `bazel vendor //...` causes a crash with the following stacktrace:

```
INFO: Analyzed 9593 targets (7882 packages loaded, 340534 targets configured).
INFO: Found 9593 targets...
INFO: Elapsed time: 21.080s, Critical Path: 0.00s
INFO: 0 processes.
INFO: Build completed successfully, 0 total actions
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.NullPointerException: Cannot invoke "com.google.devtools.build.skyframe.NodeEntry.getValue()" because "nodeEntry" is null
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.collectReposFromTargets(VendorCommand.java:310)
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.vendorTargets(VendorCommand.java:290)
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.exec(VendorCommand.java:161)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:664)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:244)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:573)
        at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$1(GrpcServerImpl.java:644)
        at io.grpc.Context$1.run(Context.java:566)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
```

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run `bazel vendor //...`. Might require a repository with many/large dependencies.

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 7.3.0

**Comment by fmeum** [comment]
@meteorcloudy

**Comment by meteorcloudy** [comment]
@mortenmj Thanks for reporting, is it possible to provide a reproducible case?

**Comment by mortenmj** [comment]
I just narrowed down which target causes the issue. We had an alias to a file:

```
# mkdir foo
# touch foo/file.txt
# echo "alias(name='foo', actual='file.txt')" > foo/BUILD.bazel
# bazel vendor --vendor_dir=~/.vendor //foo/...

FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.NullPointerException: Cannot invoke "com.google.devtools.build.skyframe.NodeEntry.getValue()" because "nodeEntry" is null
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.collectReposFromTargets(VendorCommand.java:310)
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.vendorTargets(VendorCommand.java:290)
        at com.google.devtools.build.lib.bazel.commands.VendorCommand.exec(VendorCommand.java:161)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:664)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:244)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:573)
        at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$1(GrpcServerImpl.java:644)
        at io.grpc.Context$1.run(Context.java:566)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by meteorcloudy** [comment]
Thanks! I'll take a look

**Comment by meteorcloudy** [comment]
@bazel-io fork 7.3.1

**Comment by meteorcloudy** [comment]
https://github.com/bazelbuild/bazel/pull/23301 should fix the problem, and hopefully we can get it in 7.3.1 

/cc @Wyverald

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.3.1 RC2](https://github.com/bazelbuild/bazel/releases/tag/7.3.1rc2). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.3.1rc2`. Thanks!
